### DISCUSSION TITLE
Problem with Count in queryRaw, return BigInt

### DISCUSSION THREAD

**Discussion by lucasgarcia98** [prisma/prisma#14863] [Q&A] [answered]
Hello, I have a problem after updating to version 4 of Prisma, which refers to a return of a Count using queryRaw, instead of returning an integer value, it is returning a BigInt, adding an "n" after the number, in this case referring to quantity.

Ex:
- Query: count(*) as quantity
- Return: quantity: 6n

**Comment by ludralph** [comment, marked as answer]
Hi @lucasgarcia98 👋 

From Prisma version 4 and above this is indeed expected behaviour.

If you want to remove the n suffix, you could do something like this
In this example, the count was returned as BigInt and then it was converted to number through Number constructor

```js
import { PrismaClient } from '@prisma/client';

const prisma = new PrismaClient({
  log: ['query'],
});

async function main() {
  await prisma.customers.createMany({
    data: [
      {
        country: 'USA',
        customerName: 'John Doe',
      },
      {
        country: 'USA',
        customerName: 'Jane Doe',
      },
      {
        country: 'Canada',
        customerName: 'Adams Doe',
      },
    ],
  });

  const clients = await prisma.customers.groupBy({
    by: ['country'],
    _count: true,
  });

  console.log(clients);

  const clientsWithRawQuery: any =
    await prisma.$queryRaw`SELECT Country, COUNT(country) as totalCount
FROM "public"."Customers" GROUP BY Country`;

  console.log(clientsWithRawQuery);

  // @ts-ignore
  console.log(
    'Before Conversion: Typeof Count:',
    typeof clientsWithRawQuery[0].totalcount
  );

  //Converting to Integer
  // @ts-ignore
  clientsWithRawQuery.forEach((countryObject: any) => {
    countryObject.totalcount = Number(countryObject.totalcount);
  });

  console.log(
    'After Conversion: Typeof Count:',
    typeof clientsWithRawQuery[0].totalcount
  );
}

main()
  .catch((e) => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });
```
Response:
```
prisma:query SELECT Country, COUNT(country) as totalCount FROM "public"."Customers" GROUP BY Country
[
  { country: 'USA', totalcount: 4n },
  { country: 'Canada', totalcount: 2n }
]
Before Conversion: Typeof Count: bigint
[
  { country: 'USA', totalcount: 4 },
  { country: 'Canada', totalcount: 2 }
]
After Conversion: Typeof Count: number
```

**Reply by lucasgarcia98** [reply]
Thanks for the answered, but I would like to know if this is normal to happen, or if it would be a bug related to the new version.

**Reply by ludralph** [reply]
It is normal and expected behaviour.

**Comment by montanari2019** [comment]
In other words, gambiarra to solve the problem, mercy saw

**Comment by SirS4lute** [comment]
Prisma is a horrible tool, ffs

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
